### ISSUE TITLE
Race condition to calling eval_format_fetch_in_workspace

### ISSUE THREAD

**Issue by ericphanson** [fonsp/Pluto.jl#869]
on Julia 1.5.3, with the error message:
```julia
MethodError: no method matching eval_format_fetch_in_workspace(::Tuple{Pluto.ServerSession,Pluto.Notebook}, ::Nothing, ::Base.UUID, ::Bool, ::Tuple{Set{Symbol},Set{Symbol}})
Closest candidates are:
  eval_format_fetch_in_workspace(::Union{Tuple{Pluto.ServerSession,Pluto.Notebook}, Pluto.WorkspaceManager.Workspace}, ::Expr, ::Base.UUID, ::Bool, ::Union{Nothing, Tuple}) at /home/ec2-user/.julia/packages/Pluto/luIaC/src/evaluation/WorkspaceManager.jl:211
  eval_format_fetch_in_workspace(::Union{Tuple{Pluto.ServerSession,Pluto.Notebook}, Pluto.WorkspaceManager.Workspace}, ::Expr, ::Base.UUID, ::Bool) at /home/ec2-user/.julia/packages/Pluto/luIaC/src/evaluation/WorkspaceManager.jl:211
  eval_format_fetch_in_workspace(::Union{Tuple{Pluto.ServerSession,Pluto.Notebook}, Pluto.WorkspaceManager.Workspace}, ::Expr, ::Base.UUID) at /home/ec2-user/.julia/packages/Pluto/luIaC/src/evaluation/WorkspaceManager.jl:211
```

Not sure if this is known but figured I'd report it in case it isn't.

**Comment by pankgeorg** [comment]
Hey - thanks for reporting this!
Are you using the Pluto latest latest from the master branch or the 'official' release?

**Comment by ericphanson** [comment]
Just the official release (from `Pkg.add("Pluto")`), version 0.12.18

**Comment by kangpyo** [comment]
I'm getting another error to install for 0.12.18

(@JuliaPro_v1.5.3-1) pkg> add Pluto@0.12.18
  Resolving package versions...
ERROR: Unsatisfiable requirements detected for package FuzzyCompletions [fb4132e2]:
 FuzzyCompletions [fb4132e2] log:
 ├─possible versions are: [0.1.0, 0.2.0-0.2.6, 0.3.0, 0.4.0] or uninstalled
 ├─restricted by compatibility requirements with Pluto [c3e4b0f8] to versions: [0.3.0, 0.4.0]
 │ └─Pluto [c3e4b0f8] log:
 │   ├─possible versions are: [0.2.0, 0.3.0-0.3.6, 0.4.0-0.4.3, 0.5.0-0.5.21, 0.6.0-0.6.4, 0.7.0-0.7.10, 0.8.0-0.8.10, 0.9.0-0.9.11, 0.10.0-0.10.13, 0.11.0-0.11.14, 0.12.0-0.12.18] or uninstalled
 │   └─restricted to versions 0.12.18 by an explicit requirement, leaving only versions 0.12.18
 └─restricted by compatibility requirements with Atom [c52e3926] to versions: [0.1.0, 0.2.0-0.2.6] — no versions left
   └─Atom [c52e3926] log:
     ├─possible versions are: [0.8.0-0.8.8, 0.9.0-0.9.1, 0.10.0-0.10.2, 0.11.0-0.11.3, 0.12.0-0.12.30] or uninstalled
     └─restricted to versions 0.12.25 by an explicit requirement, leaving only versions 0.12.25

or if i just add Pluto, it became 0.12.12

(@JuliaPro_v1.5.3-1) pkg> add Pluto
  Resolving package versions...
Updating `~/.julia/environments/JuliaPro_v1.5.3-1/Project.toml`
  [c3e4b0f8] + Pluto v0.12.12
Updating `~/.julia/environments/JuliaPro_v1.5.3-1/Manifest.toml`
  [99f44e22] + MsgPack v1.1.0
  [c3e4b0f8] + Pluto v0.12.12

**Comment by pankgeorg** [comment]
@kangpyo it seems your default  project, which lives at the `~/.julia/environments/JuliaPro_v1.5.3-1/` folder, has `Atom` which needs a pinned version of `FuzzyCompletions`, which we haven't tested yet. Try to either remove `Atom` or use another project:
`julia --project="."` will give you a fresh project on the `pwd` (where you run julia).

**Comment by fonsp** [comment]
Try `Pkg.update()`

**Comment by fonsp** [comment]
@ericphanson Does this happen for every notebook?  If not, can you share a notebook that hangs?

**Comment by ericphanson** [comment]
@fonsp I’ve only seen it once, and I did not see a stack trace, just that message.

When it hung and printed that error message, I just restarted the notebook and didn’t see an error when I re-ran it. So I’m not really sure what caused it or how to re-trigger it. Unfortunately I can’t share the notebook either (but it wasn’t doing anything weird with Pluto internals, just some DataFrames manipulations and things like that).

**Comment by ericphanson** [comment]
Sorry @fonsp, I was wrong, there is a stacktrace!

```julia
      From worker 2:    An error occurred (InvalidArgument) when calling the GetObject operation: Invalid version id specified                                                                                                                                                 
MethodError: no method matching eval_format_fetch_in_workspace(::Tuple{Pluto.ServerSession,Pluto.Notebook}, ::Nothing, ::Base.UUID, ::Bool, ::Tuple{Set{Symbol},Set{Symbol}})                                                                                                  
Closest candidates are:                                                                                                                                                                                                                                                        
  eval_format_fetch_in_workspace(::Union{Tuple{Pluto.ServerSession,Pluto.Notebook}, Pluto.WorkspaceManager.Workspace}, ::Expr, ::Base.UUID, ::Bool, ::Union{Nothing, Tuple}) at /home/ec2-user/.julia/packages/Pluto/luIaC/src/evaluation/WorkspaceManager.jl:211              
  eval_format_fetch_in_workspace(::Union{Tuple{Pluto.ServerSession,Pluto.Notebook}, Pluto.WorkspaceManager.Workspace}, ::Expr, ::Base.UUID, ::Bool) at /home/ec2-user/.julia/packages/Pluto/luIaC/src/evaluation/WorkspaceManager.jl:211                                       
  eval_format_fetch_in_workspace(::Union{Tuple{Pluto.ServerSession,Pluto.Notebook}, Pluto.WorkspaceManager.Workspace}, ::Expr, ::Base.UUID) at /home/ec2-user/.julia/packages/Pluto/luIaC/src/evaluation/WorkspaceManager.jl:211                                               
Stacktrace:                                                                                                                                                                                                                                                                    
 [1] run_single!(::Tuple{Pluto.ServerSession,Pluto.Notebook}, ::Pluto.Cell, ::Pluto.ReactiveNode) at /home/ec2-user/.julia/packages/Pluto/luIaC/src/evaluation/Run.jl:107                                                                                                      
 [2] run_reactive!(::Pluto.ServerSession, ::Pluto.Notebook, ::Pluto.NotebookTopology, ::Pluto.NotebookTopology, ::Array{Pluto.Cell,1}; deletion_hook::Function, persist_js_state::Bool) at /home/ec2-user/.julia/packages/Pluto/luIaC/src/evaluation/Run.jl:78                 
 [3] run_reactive! at /home/ec2-user/.julia/packages/Pluto/luIaC/src/evaluation/Run.jl:26 [inlined]                                                                                                                                                                            
 [4] macro expansion at /home/ec2-user/.julia/packages/Pluto/luIaC/src/evaluation/Run.jl:13 [inlined]                                                                                                                                                                          
 [5] (::Pluto.var"#79#81"{Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}},Pluto.ServerSession,Pluto.Notebook,Pluto.NotebookTopology,Pluto.NotebookTopology,Array{Pluto.Cell,1}})() at ./task.jl:356^C                                                     
                                                                                                                                                                                                                              
```

I scrolled way up in my tmux session to double-check, and there it was. (Tmux + scrolling is kind of glitchy for me, and I remember I was having visual bugs like repeated lines etc in the output when that error occurred, so I think that's why I missed it then).

**Comment by pankgeorg** [comment]
Eric - did you remember if this error was accompanied with a front-end error in a scary `alert`?

**Comment by ericphanson** [comment]
No, I don't remember seeing that, I just remember a simple cell wasn't executing, so I went to look at the terminal and saw that MethodError, so I restarted Pluto.

**Comment by fonsp** [comment]
This was most likely fixed by #1008. ✨ The origin of the bug is fixed in this line:

https://github.com/fonsp/Pluto.jl/commit/114673c8a3f9bb89cb1f30e821c5a95c086c2b22#diff-49609ca9d23c45546fff0029e2b450518a61925da221bd6a9a8bd59fb23b64bcL114-R114

Before #1008, the race condition in this issue meant that `cell.parsedcode` could be changed to `nothing` while `run_single!` was still running in another task. 

#1008 was originally intended to fix another race condition, #1007, so it makes sense that this race condition was fixed as well.
